Sources/ContentScopeScripts/dist/contentScope.js

```diff
@@ -3864,78 +3864,230 @@
 
       switch (type) {
       case 'hide':
-          if (!element.hidden) {
-              hideNode(element);
-          }
+          hideNode(element);
           break
       case 'hide-empty':
-          if (!element.hidden && isDomNodeEmpty(element)) {
+          if (isDomNodeEmpty(element)) {
               hideNode(element);
+              appliedRules.add(rule);
           }
           break
       case 'closest-empty':
-          // if element already hidden, continue onto parent element
-          if (element.hidden) {
-              collapseDomNode(element.parentNode, type);
-              break
+          // hide the outermost empty node so that we may unhide if ad loads
+          if (isDomNodeEmpty(element)) {
+              collapseDomNode(element.parentNode, rule, element);
+          } else if (previousElement) {
+              hideNode(previousElement);
+              appliedRules.add(rule);
           }
+          break
+      }
+  }
 
-          if (isDomNodeEmpty(element)) {
-              hideNode(element);
-              collapseDomNode(element.parentNode, type);
+  /**
+   * Unhide previously hidden DOM element if content loaded into it
+   * @param {HTMLElement} element
+   * @param {Object} rule
+   * @param {HTMLElement} [previousElement]
+   */
+  function expandNonEmptyDomNode (element, rule, previousElement) {
+      if (!element) {
+          return
+      }
+      const type = rule.type;
+
+      const alreadyHidden = hiddenElements.has(element);
+
+      switch (type) {
+      case 'hide':
+          // only care about rule types that specifically apply to empty elements
+          break
+      case 'hide-empty':
+          if (alreadyHidden && !isDomNodeEmpty(element)) {
+              unhideNode(element);
+          }
+          break
+      case 'closest-empty':
+          // iterate upwards from matching DOM elements until we arrive at previously
+          // hidden element. Unhide element if it contains visible content.
+          if (alreadyHidden) {
+              if (!isDomNodeEmpty(element)) {
+                  unhideNode(element);
+              }
+              break
+          } else {
+              expandNonEmptyDomNode(element.parentNode, rule);
           }
           break
-      default:
-          console.log(`Unsupported rule: ${type}`);
       }
   }
 
+  /**
+   * Hide DOM element
+   * @param {HTMLElement} element
+   */
   function hideNode (element) {
+      // maintain a reference to each hidden element along with the properties
+      // that are being overwritten
+      const cachedDisplayProperties = {
+          display: element.style.display,
+          'min-height': element.style.minHeight,
+          height: element.style.height
+      };
+      hiddenElements.set(element, cachedDisplayProperties);
+
+      // apply styles to hide element
       element.style.setProperty('display', 'none', 'important');
+      element.style.setProperty('min-height', '0px', 'important');
```

Comment:
I added these to address a case where the child element of a leaderboard container has `position: fixed` and a set `min-height` value, so it appears even when the parent element has `display: none`.